### PR TITLE
Fix Glue::execute to run multiple sql query string

### PR DESCRIPTION
**Pull request by ming535**
1. fix this issue https://github.com/gluesql/gluesql/issues/258
2. upgrade rust toolchain due to clippy crash on mac https://github.com/rust-lang/rust-clippy/issues/8470

### CODE REVIEW

File: core/src/glue.rs [hunk 1]
```diff
@@ -27,28 +27,39 @@ impl<T, U: GStore<T> + GStoreMut<T>> Glue<T, U> {
...
 
-        plan(storage, statement).await
+        let mut statements = Vec::<Statement>::new();
+        let parsed_iter = parsed.iter();
+        for parsed in parsed_iter {
+            let translated = translate(parsed)?;
+            let planned = plan(storage, translated).await?;
+            statements.push(planned);
+        }
+        Ok(statements)
```
Comment (ever0de) [inline core/src/glue.rs:41]:
```suggestion
        stream::iter(parsed)
            .map(|parsed| translate(&parsed))
            .then(|statement| async move { plan(storage, statement?).await })
            .try_collect()
            .await
```

we recommend fp style code
you can use `futures::Stream` to replace mutable variable
- add `futures::{TryStreamExt, StreamExt}`

File: core/src/glue.rs [hunk 2]
```diff
@@ -27,28 +27,39 @@ impl<T, U: GStore<T> + GStoreMut<T>> Glue<T, U> {
...
+    pub fn execute<Sql: AsRef<str>>(&mut self, sql: Sql) -> Result<Vec<Payload>> {
+        let statements = block_on(self.plan(sql))?;
 
-        self.execute_stmt(statement)
+        let mut payloads = Vec::<Payload>::new();
+        for statement in statements.iter() {
+            let payload = self.execute_stmt(statement)?;
+            payloads.push(payload);
+        }
+        Ok(payloads)
```
Comment (ever0de) [inline core/src/glue.rs:56]:
```suggestion
        statements
            .iter()
            .map(|statement| self.execute_stmt(statement))
            .collect()
```
we recommend fp style code 22 🚀

File: core/src/glue.rs [hunk 3]
```diff
@@ -62,9 +73,14 @@ impl<T, U: GStore<T> + GStoreMut<T>> Glue<T, U> {
...
+    pub async fn execute_async<Sql: AsRef<str>>(&mut self, sql: Sql) -> Result<Vec<Payload>> {
+        let statements = self.plan(sql).await?;
 
-        self.execute_stmt_async(statement).await
+        let mut payloads = Vec::<Payload>::new();
+        for statement in statements.iter() {
+            let payload = self.execute_stmt_async(statement).await?;
+            payloads.push(payload);
+        }
+        Ok(payloads)
```
Comment (ever0de) [inline core/src/glue.rs:84]: Add a blank 👍 

```suggestion
        }
        
        Ok(payloads)
```

File: storages/sled-storage/tests/sled_transaction.rs [hunk 4]
```diff
@@ -30,16 +30,25 @@ macro_rules! exec {
...
-        assert_eq!($glue.execute($sql), $result);
+        let res = $glue.execute($sql);
+        let expeced_result: Result<Payload> = $result;
+        if let Ok(r) = res {
+            assert_eq!(expeced_result.is_ok(), true);
+            assert_eq!(r.len(), 1);
+            assert_eq!(r[0], expeced_result.unwrap());
+        } else {
+            assert_eq!(res.err(), expeced_result.err());
+        }
```
Comment (ever0de) [inline storages/sled-storage/tests/sled_transaction.rs:41]:
```suggestion
        let actual = $glue.execute($sql);
        assert_eq!(actual, $expected.map(|payload| vec![payload]));
```

Let's change it simply!

File: storages/sled-storage/tests/sled_transaction.rs [hunk 5]
```diff
@@ -30,16 +30,25 @@ macro_rules! exec {
...
 }
 
 macro_rules! test_idx {
     ($glue: ident $sql: literal, $idx: expr, $result: expr) => {
-        let statement = $glue.plan($sql).await.unwrap();
-
-        test_indexes(&statement, Some($idx));
-        assert_eq!($glue.execute_stmt(statement), $result);
+        let statements = $glue.plan($sql).await.unwrap();
+        assert_eq!(statements.len(), 1);
```
Comment (ever0de) [inline storages/sled-storage/tests/sled_transaction.rs:48]: Good 👍

**Review by ever0de — COMMENTED** [review]
I left a few reviews.

GlueSQL is oriented towards functional-programming, so let's try to code in fp style as much as possible.

Thanks

File: examples/hello_world.rs [hunk 6]
```diff
@@ -4,6 +4,7 @@ mod hello_world {
         prelude::{Glue, Payload, Value},
         sled_storage::SledStorage,
     };
+    use std::fs;
```
[the remark this answers is not in the document]
Comment (ming535) [reply examples/hello_world.rs:7]: I don't think so..., `{...}` is in the `gluesql::` namespace.

File: core/src/glue.rs (hunk 3 above)
Comment (ming535) [reply core/src/glue.rs]: done

File: storages/sled-storage/tests/sled_transaction.rs (hunk 4 above)
Comment (ming535) [reply storages/sled-storage/tests/sled_transaction.rs]: done

File: examples/hello_world.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (ever0de) [reply examples/hello_world.rs:7]:
```rust
    use {
        gluesql::{
            prelude::{Glue, Payload, Value},
            sled_storage::SledStorage,
        },
        std::fs,
    };
```

It seems to be possible by grouping the top in this way.

File: cli/src/print.rs [hunk 7]
```diff
@@ -13,7 +13,11 @@ impl<W: Write> Print<W> {
         Print { output }
     }
 
-    pub fn payload(&mut self, payload: Payload) -> Result<()> {
+    pub fn payloads(&mut self, payloads: Vec<Payload>) -> Result<()> {
+        payloads.iter().try_for_each(|p| self.payload(p))
+    }
```
Comment (ever0de) [inline cli/src/print.rs:18]:
```suggestion
    pub fn payloads(&mut self, payloads: &[Payload]) -> Result<()> {
        payloads.iter().try_for_each(|payload| self.payload(payload))
    }
```

good 👍 
Learned `try_for_each`! thank you.

The `Vec<Payload>` variable might be ok to borrow here. Try this `&[Payload]`

**Review by ever0de — APPROVED** [review]
If one comment is resolved, Looks good to me

File: cli/src/print.rs (hunk 7 above)
Comment (ming535) [reply cli/src/print.rs]: done. It seems that clippy cannot found this issue when I change it the first time...

File: core/src/glue.rs [hunk 8]
```diff
@@ -9,6 +10,7 @@ use {
         translate::translate,
     },
     futures::executor::block_on,
+    futures::stream::{self, StreamExt},
```
Comment (ever0de) [inline core/src/glue.rs:13]:
```suggestion
    futures::stream::{self, StreamExt, TryStreamExt},
```

Please move the ones that are not grouped above thanks !

**Review by panarch — APPROVED** [review]
Now finally `Glue#execute` has the right behavior.
Thanks a lot! Awesome, merging to the main
👍 👍 👍 

Also thanks to the good detailed review @ever0de

**Pull request merged**